### PR TITLE
refactor DatasourceBuilder.withConnectionTimeout

### PR DESCRIPTION
**Pull request by stephane-airbyte**
simply moving changing the type of connectionTimeoutMs form `long` to `Duration`. The name of the variable also becomes connectionTimeout

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33439?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #33357** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33357?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #33358** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33358?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #33270** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33270?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #33439** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33439?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
          * **PR #33440** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33440?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #33441** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33441?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/33439?utm_source=stack-comment).

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/factory/DataSourceFactory.java [hunk 1]
```diff
@@ -180,7 +180,8 @@ private static class DataSourceBuilder {
     private String jdbcUrl;
     private int maximumPoolSize = 10;
     private int minimumPoolSize = 0;
-    private long connectionTimeoutMs;
+    // This feels really weird, but it's strictly equivalent to the original long connectionTimeoutMs
```
Comment (akashkulk) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/factory/DataSourceFactory.java:183]: Can we just remove this comment?

**Review by akashkulk — APPROVED** [review]

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/factory/DataSourceFactory.java (hunk 1 above)
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/factory/DataSourceFactory.java:183]: why ? I do genuinely think having a default set to 0 is weird

Comment (postamar) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/factory/DataSourceFactory.java:183]: I think that's worth commenting too but I don't think referencing "the original long connectionTimeoutMs" is going to be useful to the reader.

**Review by postamar — APPROVED** [review]

**Pull request merged**